### PR TITLE
Fixes #17588 - Remove 1.16 deprecations

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

File: app/models/setting.rb [hunk 1]
```diff
@@ -103,17 +103,6 @@ def self.method_missing(method, *args)
     super
   rescue NoMethodError
```
Comment (domcleal) [inline app/models/setting.rb]: No need to keep this empty method.

**Review by domcleal — CHANGES_REQUESTED** [review]
Please ensure the tests pass.

**Comment by dLobatog** [comment]
Tests passing again

File: test/models/settings/email_setting_test.rb [hunk 2]
```diff
@@ -1,72 +1,29 @@
...
-      assert_equal("example@example.com", Setting[:email_reply_address])
-    end
-
-    test 'delivery_method setting value can be a string' do
-      @delivery_settings["delivery_method"] = 'smtp'
-      load_defaults
-      assert_equal(@delivery_settings["smtp_settings"], Setting::Email.delivery_settings)
-    end
+  setup do
+    Setting::Email.stubs(:mailconfig).returns({})
```
Comment (domcleal) [inline test/models/settings/email_setting_test.rb]: This method's being removed, no need to stub it now.

File: test/models/settings/email_setting_test.rb [hunk 3]
```diff
@@ -1,72 +1,29 @@
...
-      Setting::Email.stubs(:mailconfig).returns({})
-    end
-
-    test 'email settings should be editable if email.yaml not exist' do
-      Setting[:delivery_method] = :smtp
-      Setting[:smtp_address] = "sm@example.com"
-      Setting[:smtp_authentication] = "plain"
-      assert_equal({"address" => "sm@example.com", "authentication" => "plain"}, Setting::Email.delivery_settings)
-    end
+  test 'email settings should be editable if email.yaml not exist' do
```
Comment (domcleal) [inline test/models/settings/email_setting_test.rb]: No need to mention email.yaml here, it's obsolete. (Suggest renaming this test to simply say it tests the output of `delivery_settings` when settings are modified.)

**Review by domcleal — CHANGES_REQUESTED** [review]
There are some more `email.yaml` leftovers - please remove the template file, and entries in `.gitignore` and `config/spring.rb`.

**Comment by dLobatog** [comment]
Updated- thanks for the review @domcleal

**Review by domcleal — APPROVED** [review]

**Comment by domcleal** [comment]
Merged, thanks @dLobatog.

**Pull request merged**